src/components/CardDetails.js

```diff
@@ -0,0 +1,270 @@
+            <div className="relative">
+              <div
+                onClick={() => {}}
+                className="h-48 w-48 m-2 gap-4 bg-no-repeat relative z-50"
+                style={{
+                  backgroundImage: `url("https://dokkan.wiki/assets/global/en/character/thumb/card_${characterThumb}_thumb.png")`,
+                  backgroundSize: `100%`,
+                }}
+              ></div>
+              <img className={characterRarityPosition} src={characterRarity} />
+              <img
+                className="w-40 absolute top-[15px] right-[25px] z-0"
+                src={characterType}
+              />
+              <img
+                className="w-20 absolute top-[-12px] right-[4px] z-50"
+                src={characterSuperOrExtreme}
+              />
+            </div>
+          )}
+        </div>
+      </div>
+      <div className="bg-gradient-radial from-purple-200 via-purple-100 to-purple-50 m-1 p-2 shadow-[inset_0_-5px_6px_rgba(0,0,0,0.6)] border-2 border-slate-900 w-[85%] h-min">
+        <p>Leader Skill: {cardDetails.ls_description}</p>
+      </div>
+
+      <div className="bg-gradient-radial from-purple-200 via-purple-100 to-purple-50 m-1 p-2 shadow-[inset_0_-5px_6px_rgba(0,0,0,0.6)] border-2 border-slate-900 w-[85%] h-min">
+        <p>
+          {cardDetails.sa_name}: {cardDetails.sa_description}
+        </p>
+      </div>
+      <div>
+            {Auth.loggedIn() ? (
+              <div>
+                {userCharacters.includes(cardDetails.id) ? (
+                  <div className="bg-orange-300 rounded-md border-2 border-slate-900 flex 2xl:h-12 2xl:mt-5">
+                    <button
+                      onClick={() => {
+                        handleRemoveCharacter();
+                      }}
+                      className="relative hover:bg-orange-400 m-0.5 pr-10 pl-10 pt-2 pb-2"
+                    >
+                      Remove Character
+                    </button>
+                  </div>
+                ) : (
+                  <div className="bg-orange-300 rounded-md border-2 border-slate-900 flex 2xl:h-12 2xl:mt-5">
+                  <button
+                    onClick={() => {
+                      handleSaveCharacter();
+                    }}
+                    className="relative hover:bg-orange-400 m-0.5 pr-10 pl-10 pt-2 pb-2"
+                  >
+                    Add character
+                  </button>
+                  </div>
+                )}
+              </div>
+            ) : (
+              <div>
+                <h1>Log in to add characters</h1>
+              </div>
+            )}
+          </div>
+
+      <div className="grid grid-cols-2">
+        <div className="m-4 h-full max-h-[60vh]">
+          <h1>Character Links:</h1>
+          <div className="border-2 border-black">
+            {characterLinks &&
+              characterLinks.map((linkText) => {
+                return (
+                  <div>
```

Comment:
This div should have a key in case react optimizes a rerender if the character links change. I think you can use the linkText as the key